developers_chamber/project_utils.py

```diff
@@ -108,3 +114,49 @@ def compose_install(project_name, compose_files, var_dirs=None, containers_dir_t
 
     for container_name, command in install_container_commands or ():
         compose_run(project_name, compose_files, [container_name], command)
+
+
+def start_task(jira_url, jira_username, jira_api_key, toggl_api_key, project_key, issue_key):
+    running_timer = get_running_timer_data(toggl_api_key)
+    if running_timer:
+        if confirm('Timer is already running do you want to log it?'):
+            stop_task(jira_url, jira_username, jira_api_key, toggl_api_key)
+
+    issue_key = clean_issue_key(issue_key, project_key)
+    issue_data = get_issue_fields(jira_url, jira_username, jira_api_key, issue_key, project_key)
+    toggl_description = '{} {}'.format(issue_key, issue_data.summary)
+    start_timer(toggl_api_key, toggl_description)
+    return 'Toggle was started with description "{}"'.format(toggl_description)
+
+
+def stop_task(jira_url, jira_username, jira_api_key, toggl_api_key):
+    running_timer = get_running_timer_data(toggl_api_key)
+    if running_timer:
+        match = re.match('(?P<issue_key>.{3}-\d+).*', running_timer.description)
+        if match:
+            issue_key = match.group('issue_key')
+            get_issue_fields(jira_url, jira_username, jira_api_key, issue_key)
+            stopped_timer = stop_running_timer(toggl_api_key)
+            log_issue_time(
+                jira_url,
+                jira_username,
+                jira_api_key,
+                issue_key,
+                time_spend=timedelta(seconds=stopped_timer.duration),
+                comment='Toggl {}'.format(stopped_timer.id),
+            )
+            return 'Timner was stopped and time was logged'
+        else:
+            ClickException('Invalid running task description')
+    else:
+        ClickException('No running task')
+
+
+def create_or_update_pull_request(jira_url, jira_username, jira_api_key, bitbucket_username, bitbucket_password,
+                                  bitbucket_destination_branch_name, bitbucket_repository_name):
+    issue_key = clean_issue_key()
+    issue_data = get_issue_fields(jira_url, jira_username, jira_api_key, issue_key)
+    return bitbucket_create_pull_request(
+        bitbucket_username, bitbucket_password, '{} {}'.format(issue_key, issue_data.summary), issue_data.description,
+        get_current_branch_name(), bitbucket_destination_branch_name, bitbucket_repository_name
+    )
```

Comment:
stále se rozhoduji zda přenášet description z issue, atlassina je totiž skvělý v tom,že má všude stejný wysiwyg editor (to je ironie)...

Reply:
I wouldn't do that, that's just duplication, isn't it? You can easily get to issue by clicking on its identifier in the pull request title.